packages/core/src/shared/extensions/ssh.ts

```diff
@@ -148,3 +149,35 @@ export async function startVscodeRemote(
 
     await new ProcessClass(vscPath, ['--folder-uri', workspaceUri]).run()
 }
+/**
+ * Uses provided sshPath to get version of OpenSSH. If sshPath is not OpenSSH, returns undefined.
+ * @param sshPath
+ * @returns
+ */
+export async function getSshVersion(sshPath: string): Promise<string | undefined> {
+    const result = await new ChildProcess(sshPath, ['-V'], { collect: true }).run()
+
+    return parseSshVersion(result.stdout === '' ? result.stderr : result.stdout)
+}
+
+export async function ensureSshVersionGte(sshPath: string, minVersion: string): Promise<void | never> {
+    const sshVersion = await getSshVersion(sshPath)
+    if (sshVersion && semver.lt(sshVersion, minVersion)) {
+        const msg = `SSH version ${sshVersion} is not supported, please upgrade OpenSSH to ${minVersion} or higher`
+        getLogger().error(msg)
+        throw new Error(msg)
+    }
+}
+
+function parseSshVersion(output: string): string | undefined {
+    const match = output.match(/OpenSSH_(\d+)\.(\d+)/)
```

Comment:
The error is coming from the connect script, i.e. the child process that launches the new instance of VSCode so I am not sure how to surface it. Especially since the child process is successfully opening VSCode, but then the newly opened VSCode instance throws an error. We would have to route that error back to this instance of VSCode somehow.

Reply:
Yes, when we invoke `code` with a remote URL, vscode invokes `ssh` implicitly for us. But what I'm thinking is that we can manually invoke `ssh` ourselves, before invoking `code`, and just run a trivial command on the remote like `sh -c 'true'`. That gives a true picture of whether ssh can work.

If that fails, we can report/log the ssh output and that gives us valuable troubleshooting info. Without needing to check for specific ssh versions. This is also future-proof for any changes we make to our .ssh/config.

Reply:
Ahhh, I see. That's a good idea. I'll work on that in a fresh PR and link it here.